### DISCUSSION TITLE
Having two different types of trials intermixed

### DISCUSSION THREAD

**Discussion by Altr37** [jspsych/jsPsych#2028] [Q&A]
Hi everyone,

I currently have a code working with sentences, followed by comprehension questions. (Copied below)
I want to change it, such that only some of the sentences will be followed by a question, while others will not, and will proceed to the next trial. I want the order to be intermixed between these two types of trials. 
How can I do that?

code:
```js
 var sentence = {
                     type: 'html-keyboard-response',
                     stimulus: function() {
                         return '<span style="font-size:25px;">' + jsPsych.timelineVariable('sentence', true) + '</span>';
                     },
                     choices: ['space'],
                     post_trial_gap: 500,
                     prompt: '<br><br> Press the space bar to continue',

                 };

                 var question = {
                     type: 'html-keyboard-response',
                     stimulus: function() {
                         return '<span style="font-size:25px;">' + jsPsych.timelineVariable('question', true) + '</span>';
                     },
                     choices: ['f','j'],
                     post_trial_gap: 1000,
                     prompt: '<br><br> NO - <strong>F</strong>  &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;  &nbsp; &nbsp; &nbsp; &nbsp; YES - <strong>J</strong>',
                     data: {
                         sentence: jsPsych.timelineVariable('sentence'),
                         question: jsPsych.timelineVariable('question'),
                         condition: jsPsych.timelineVariable('condition'),
                         correct_response: jsPsych.timelineVariable('correct_response'),
                         task_part: jsPsych.timelineVariable('task_part'),
                         trial_part: 'question',
                      },
                     on_finish: function(data) {
                         var acc = false;
                         if (data.correct_response == jsPsych.pluginAPI.convertKeyCodeToKeyCharacter(data.key_press)) {
                             acc = true;
                         }
                         data.accuracy = acc;
                     }
                 };
```
The materials currently look like that:
```js
{
sentence: "The bla bla.", 
question: "Was it the bla bla?", 
correct_response: "f", 
condition: "filler", 
task_part: "sentences", 
},
```
Instead, I want some trials to have a 'question' part, and others not.

I will appreciate any help!
Thank you!

**Comment by becky-gilbert** [comment]
Hi @Altr37, what you can do is put the question trial inside of a [conditional node](https://www.jspsych.org/overview/timeline/#conditional-timelines), and in the conditional function, check the timeline variables to see whether or not a question should be shown on the current trial. This means that you also need to put some kind of indication in your timeline variables objects about whether or not each trial should have a question. One way to do this is to set the `question` property to `null` when you don't want a question. Then your code would look like this:
```js
var sentence = {
  // parameters...
}

var question = {
  // parameters...
}

var question_conditional = {
  timeline: [question],
  conditional_function: function() {
    // check whether or not the question trial should be presented
    if (jsPsych.timelineVariable('question') == null) {
      return false;
    } else {
      return true;
    }
  }
} 

var trial_procedure = {
  timeline: [sentence, question_conditional],
  timeline_variables: your_timeline_variables_array
}
```
And your timeline variables array would look like this:
```js
 {
  sentence: "The bla bla.", 
  question: "Was it the bla bla?", // present this question after the sentence
  // more variables
},
{
  sentence: "Another trial.", 
  question: null, // no question after this sentence
  // more variables
},
```
Does that work for you?

**Reply by Altr37** [reply]
Hi,

Thank you for your reply.
I tried this, but instead of presenting no question, it currently presents 'null' as the question.

I'm copying the way I defined the sentences with no question:

sentence: "The bla bla.",
question: null,
correct_response: "j",
condition: "v4",
task_part: "sentences",


and the code:
 var question_conditional = {
                   timeline: [question],
                   conditional_function: function() {
                     // check whether or not the question trial should be presented
                     if (jsPsych.timelineVariable('question') == null) {
                       return false;
                     } else {
                       return true;
                     }
                   }
                 }


                 var sentence_procedure = {
                     timeline: [sentence, question_conditional],
                     timeline_variables: main_sent_info,
                     randomize_order: true
                 };
                 timeline.push(sentence_procedure);

**Reply by Altr37** [reply]
I've also tried putting "null" in quotation marks in both places, or have a blank in the stimuli for question part, and still get the same.

**Reply by Altr37** [reply]
I think the problem might be in accessing the correct part. So it's question.question
But I tried this as well, and it didn't work...
